### PR TITLE
Added support for BeOfType and NotBeOfType for Value type objects  #459

### PR DESCRIPTION
**Pull request by DennisNerush**
Fixing issue #459

### CODE REVIEW

File: Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs [hunk 1]
```diff
@@ -261,7 +261,83 @@ public void When_object_type_is_exactly_equal_to_the_specified_type_it_should_no
...
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () => valueTypeObject.Should().BeOfType(typeof(double));
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>();
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs]: Please also include some checks on the message so it's clear that the expected and actual types are included.

**Comment by DennisNerush** [comment]
added the required assert

**Comment by dennisdoomen** [comment]
> Added the required assert

You added it on the 2nd spec. Was that on purpose?

**Comment by DennisNerush** [comment]
what do you mean? I've added it the ObjectAssertionSpec, in the test that you asked

**Comment by DennisNerush** [comment]
My bad :) take a look now

**Pull request merged**